integration-tests/tests/src/tests/objects.ts

```diff
@@ -586,6 +606,113 @@ describe("Realm.Object", () => {
         const persons = this.realm.objects(PersonWithId);
         expect(persons.length).equals(1);
       });
+
+      describe("applying 'UpdateMode' recursively", () => {
+        let aliceId: Realm.BSON.ObjectId;
+        let bobId: Realm.BSON.ObjectId;
+        let maxId: Realm.BSON.ObjectId;
+
+        beforeEach(function (this: RealmContext & Mocha.Context) {
+          aliceId = new Realm.BSON.ObjectId();
+          bobId = new Realm.BSON.ObjectId();
+          maxId = new Realm.BSON.ObjectId();
+
+          // Create two mutual friends (Alice and Bob) and one
+          // that will be added as a friend later (Max).
+          this.realm.write(() => {
+            const alice = this.realm.create<ICollectionObject>(CollectionSchema.name, {
+              _id: aliceId,
+              name: "Alice",
+            });
+            const bob = this.realm.create<ICollectionObject>(CollectionSchema.name, {
+              _id: bobId,
+              name: "Bob",
+            });
+            const max = this.realm.create<ICollectionObject>(CollectionSchema.name, {
+              _id: maxId,
+              name: "Max",
+            });
+            // Make them mutual friends.
+            alice.list.push(bob);
+            bob.list.push(alice);
+          });
+        });
+
+        it("can be updated recursively in lists", function (this: Mocha.Context & RealmContext) {
+          this.realm.write(() => {
+            this.realm.create(
+              CollectionSchema.name,
+              {
+                _id: aliceId,
+                name: "UpdatedAlice",
+                list: [
+                  {
+                    _id: bobId,
+                    name: "UpdatedBob",
+                    list: [
+                      {
+                        _id: maxId,
+                        name: "UpdatedMax",
+                      },
+                    ],
+                  },
+                ],
+              },
+              Realm.UpdateMode.All,
+            );
+          });
+        });
+
+        it("can be updated recursively in dictionaries", function (this: Mocha.Context & RealmContext) {
+          this.realm.write(() => {
+            this.realm.create(
+              CollectionSchema.name,
+              {
+                _id: aliceId,
+                name: "UpdatedAlice",
+                dictionary: {
+                  bob: {
+                    _id: bobId,
+                    name: "UpdatedBob",
+                    dictionary: {
+                      max: {
+                        _id: maxId,
+                        name: "UpdatedMax",
+                      },
+                    },
+                  },
+                },
+              },
+              Realm.UpdateMode.All,
+            );
+          });
+        });
+
+        it("can be updated recursively in sets", function (this: Mocha.Context & RealmContext) {
```

Comment:
That's amazing that this works for Sets and Dictionaries now.  
Should we also add assertions for the updated values?  We are just testing that this doesn't throw, should we also test that it does throw?

Reply:
Great!  Thanks 🎉